src/views/htmlcontent/src/js/components/app.component.spec.ts

```diff
@@ -66,54 +236,470 @@ describe('AppComponent', function (): void {
+                expect(results.className.indexOf('hidden')).not.toEqual(-1);
+                done();
+            }, 100);
+        });
+
+        it('event toggle message pane', (done) => {
+            let dataService = <MockDataService> fixture.componentRef.injector.get(DataService);
+            let shortcutService = <MockShortcutService> fixture.componentRef.injector.get(ShortcutService);
+            spyOn(shortcutService, 'buildEventString').and.returnValue('');
+            spyOn(shortcutService, 'getEvent').and.returnValue(Promise.resolve('event.toggleMessagePane'));
+            dataService.sendWSEvent(batch2);
+            dataService.sendWSEvent(completeEvent);
+            fixture.detectChanges();
+            let messages = <HTMLElement> ele.querySelector('#messages');
+            let keyboardEvent = document.createEvent('KeyboardEvent');
+            let initMethod = typeof keyboardEvent.initKeyboardEvent !== 'undefined' ? 'initKeyboardEvent' : 'initKeyEvent';
+
+            keyboardEvent[initMethod](
+                            'keydown', // event type : keydown, keyup, keypress
+                                true, // bubbles
+                                true, // cancelable
+                                window, // viewArg: should be window
+                                false, // ctrlKeyArg
+                                false, // altKeyArg
+                                false, // shiftKeyArg
+                                false, // metaKeyArg
+                                40, // keyCodeArg : unsigned long the virtual key code, else 0
+                                0 // charCodeArgs : unsigned long the Unicode character associated with the depressed key, else 0
+            );
+            ele.dispatchEvent(keyboardEvent);
+            setTimeout(() => {
+                fixture.detectChanges();
+                expect(messages).not.toBeNull('message pane is not visible');
+                expect(messages.className.indexOf('hidden')).not.toEqual(-1);
+                done();
+            }, 100);
+        });
+
+        it('event copy selection', (done) => {
+            let dataService = <MockDataService> fixture.componentRef.injector.get(DataService);
+            let shortcutService = <MockShortcutService> fixture.componentRef.injector.get(ShortcutService);
+            spyOn(shortcutService, 'buildEventString').and.returnValue('');
+            spyOn(shortcutService, 'getEvent').and.returnValue(Promise.resolve('event.copySelection'));
+            spyOn(dataService, 'copyResults');
+            dataService.sendWSEvent(batch1);
+            dataService.sendWSEvent(completeEvent);
+            fixture.detectChanges();
+            let keyboardEvent = document.createEvent('KeyboardEvent');
+            let initMethod = typeof keyboardEvent.initKeyboardEvent !== 'undefined' ? 'initKeyboardEvent' : 'initKeyEvent';
+
+            keyboardEvent[initMethod](
```

Comment:
Can we simplify this since a bunch of tests are calling it? E.g. if we only have 1-3 params to actually configure, put this in its own function that does the right thing and shortens this test code / uses function & param names to clarify intent